### DISCUSSION TITLE
Animate element when in viewport with isVisible?

### DISCUSSION THREAD

**Discussion by jesperlandberg** [idiotWu/smooth-scrollbar#354] [Q&A] [answered]
Hi, thanks for a great plugin.

Is it possible to get an example on how I would animate elements when they come into the viewport? I Guess i use isVisible but I can't get it to work.

```
[...document.querySelectorAll('.animate-this')].forEach((el) => {
 scrollbar.isVisible(el) ? el.classList.add('is-visible') : false 
})
```

**Comment by dolphin-wood** [comment, marked as answer]
Here's an old demo about isVisible method: http://jsbin.com/noxoho/edit?js

Hope it could help you :)

**Comment by jesperlandberg** [comment]
@idiotWu thanks, perfect=)

**Comment by adamwright32** [comment]
@idiotWu This only fires when you interact with the scrollbar, what about if the element is at the top of the page or partially showing? try to get some functionality similar to waypoint.js

**Comment by dolphin-wood** [comment]
@adamwright32 I don't understand what you mean by "interact with the scrollbar". The `scrollbar.isVisible()` method  just simply compares the bounding rect of the given element with the bounding rect of scrollbar: https://github.com/idiotWu/smooth-scrollbar/blob/8ff9227a73691a937c855ec135db5d2c76d6a868/src/geometry/is-visible.ts#L3-L14

**Comment by adamwright32** [comment]
@idiotWu but thats only triggered when you are listening to the scrollbar. So if the page loads and you don't scroll it won't fire. Or maybe i'm doing it wrong

```
scrollbar.addListener(function (status) {
            // Content Feature Header
            $(".js-header").each(function () {
                var $this = $(this);

                if(scrollbar.isVisible(this))
                {
                    $this.addClass("animated titleInRight");
                }
            });
```

**Comment by dolphin-wood** [comment]
@adamwright32 no you don't have to put it in the scroll listener. It's a static check and you can run it everywhere.

**Comment by dtab428** [comment]
@idiotWu 

Is there a way for scrollbar.isVisible(element) to check if the element isVisible on page load _without scrolling_?

Reference example: http://jsbin.com/hujugirexi/edit?html,js,output (I re-ordered the divs in your previous example, such that "content" is on top). "content" only shows "visible: true" when scrolling at least once (even though it is visible upon page load).

Thanks for the help.

**Comment by dolphin-wood** [comment]
@dtab428 simply call `isVisible()` at anywhere you want

**Comment by vm84** [comment]
Hi, can you please give us an example on windows load?
